extensions/libarchive/CompressContent.cpp

```diff
@@ -191,14 +194,20 @@ void CompressContent::processFlowFile(const std::shared_ptr<core::FlowFile>& flo
     if (compressMode_ == CompressionMode::Compress) {
       session->putAttribute(result, core::SpecialFlowAttribute::MIME_TYPE, mimeType);
       if (updateFileName_) {
+        if (encapsulateInTar_) {
+          fileName = fileName + TAR_EXT;
+        }
         fileName = fileName + fileExtension;
         session->putAttribute(result, core::SpecialFlowAttribute::FILENAME, fileName);
       }
     } else {
       session->removeAttribute(result, core::SpecialFlowAttribute::MIME_TYPE);
       if (updateFileName_) {
-        if (fileName.size() >= fileExtension.size() && fileName.compare(fileName.size() - fileExtension.size(), fileExtension.size(), fileExtension) == 0) {
+        if (utils::StringUtils::endsWith(fileName, fileExtension)) {
           fileName = fileName.substr(0, fileName.size() - fileExtension.size());
+          if (encapsulateInTar_ && utils::StringUtils::endsWith(fileName, TAR_EXT)) {
```

Comment:
using the `encapsulateInTar_` during decompression seems strange, shouldn't we infer the need for "de-tar-ing" only from the extension?

Reply:
Currently the "de-tar-ing" depends on the `Encapsulate in TAR` property in the processor, so if that is not set in decompression only the "gunzip" operation will be run on the data and the end content will still be encapsulated in tar. That is why I followed this same process in the filename changes.

Reply:
oh I see, that makes sense